pennylane/transforms/decompose.py

```diff
@@ -214,6 +213,42 @@ def eval(self, jaxpr: "jax.core.Jaxpr", consts: Sequence, *args) -> list:
             for const, constvar in zip(consts, jaxpr.constvars, strict=True):
                 self._env_map[constvar] = const
 
+            if qml.decomposition.enabled_graph():
+
+                operations = []
+
+                # Evaluate the jaxpr of the dynamic decomposition
+                # and create a decomposition graph from the known operations.
+                for eq in jaxpr.eqns:
+
+                    # Create the graph creation if eq
+                    # - hasn't been registered yet,
+                    # - is a primitive operator, or
+                    # - has a DropVar output/invars
+                    if (
+                        not self._primitive_registrations.get(eq.primitive, None)
+                        and getattr(eq.primitive, "prim_type", "") == "operator"
+                        and eq.outvars[0].__class__.__name__ == "DropVar"
+                        and all(invar.__class__.__name__ != "Var" for invar in eq.invars)
+                    ):
+                        invals = (self.read(invar) for invar in eq.invars)
+                        with qml.QueuingManager.stop_recording():
+                            op = eq.primitive.impl(*invals, **eq.params)
+
+                        # Construct the graph for the inner operations of structured rules
+                        if not op.has_plxpr_decomposition:
```

Comment:
Like I mentioned in a previous comment, I don't think `op.has_plxpr_decomposition` has anything to do with whether we add this op to the graph. `has_plxpr_decomposition` checks the existence of a `compute_qfunc_decomposition`, which is just a qfunc version of the original `compute_decomposition`. In a sense, every single operator has a "qfunc decomposition" now, because in the new system, all decomposition rules are defined as quantum functions.